libs/s3/s3_versioning.py

```diff
@@ -114,3 +114,49 @@ def delete_object_version(self, bucket: str = None, key: str = None,
         LOGGER.debug("DELETE Object with version id response: %s", response)
 
         return response
+
+    def get_obj_tag_ver(self, bucket_name: str = None, object_name: str = None,

```

Comment:
WE don't need  test lib functions for tag versions.  You may want to delete the functions put in test libs.  We want to get away from the pattern of handling the exception and then re raising it  without any action taken in except clause. Also if you just want to handle exception to adapt to use CTFailOn for negative tests only then you can have a corresponding function in test lib only.

Reply:
@seagate-sarang-sawant We have been following the pattern laid down in existing S3 test libs. Refactoring the design will need an overhaul of other existing S3 libs as well, which is beyond the scope of the current task. I would suggest we have a detailed discussion of the expected design and we take it up it in the planned refactoring set of tasks that are coming up. cc: @sonalk0209